tests/Unit/CrudPanel/CrudPanelReadTest.php

```diff
@@ -385,4 +387,163 @@ public function testGetRelatedEntriesAttributesFromHasOneWithAcessor()
+        $this->assertCount(2, $this->crudPanel->getOperationSetting('pageLengthMenu'));
+        $this->assertTrue(in_array(40, $this->crudPanel->getOperationSetting('pageLengthMenu')[0]));
+        $this->assertEquals($this->crudPanel->getOperationSetting('pageLengthMenu')[1], ['v', 't', 'q']);
+    }
+
+    /**
+     * Tests define paginator length with and 'all' options as -1 as defined in previous versions of BP.
+     *
+     * @group
+     */
+    public function testCrudPanelPaginatorWithAllAsOption()
+    {
+        $this->crudPanel->setModel(User::class);
+        $this->crudPanel->setOperation('list');
+        $this->crudPanel->setPageLengthMenu([-1 => 'All']);
+        $this->assertCount(2, $this->crudPanel->getOperationSetting('pageLengthMenu'));
+        $this->assertTrue(in_array(-1, $this->crudPanel->getOperationSetting('pageLengthMenu')[0]));
+
+        $this->crudPanel->setPageLengthMenu([-1]);
+        $this->assertCount(2, $this->crudPanel->getOperationSetting('pageLengthMenu'));
+        $this->assertTrue(in_array(-1, $this->crudPanel->getOperationSetting('pageLengthMenu')[0]));
+
+        $this->crudPanel->setPageLengthMenu(-1);
+        $this->assertCount(2, $this->crudPanel->getOperationSetting('pageLengthMenu'));
+        $this->assertTrue(in_array(-1, $this->crudPanel->getOperationSetting('pageLengthMenu')[0]));
+    }
+
+    /**
+     * Tests define paginator length with and 'all' options as -1 as defined in previous versions of BP.
+     *
+     * @group
+     */
+    public function testCrudPanelPaginatorWithZeroAsOption()
+    {
+        $this->crudPanel->setModel(User::class);
+        $this->crudPanel->setOperation('list');
+
+        $this->crudPanel->setPageLengthMenu([0 => 'v', 30 => 't', 40 => 'q']);
+
+        $this->assertCount(2, $this->crudPanel->getOperationSetting('pageLengthMenu'));
+        $this->assertTrue(in_array(0, $this->crudPanel->getOperationSetting('pageLengthMenu')[0]));
+        $this->assertTrue(in_array('v', $this->crudPanel->getOperationSetting('pageLengthMenu')[1]));
+
+        $this->crudPanel->setPageLengthMenu([0 => 'All']);
+        $this->assertCount(2, $this->crudPanel->getOperationSetting('pageLengthMenu'));
+        $this->assertTrue(in_array(0, $this->crudPanel->getOperationSetting('pageLengthMenu')[0]));
+
+        $this->crudPanel->setPageLengthMenu([0 => 'All']);
+        $this->assertCount(2, $this->crudPanel->getOperationSetting('pageLengthMenu'));
+        $this->assertTrue(in_array(0, $this->crudPanel->getOperationSetting('pageLengthMenu')[0]));
+
+        $this->crudPanel->setPageLengthMenu([0]);
+        $this->assertCount(2, $this->crudPanel->getOperationSetting('pageLengthMenu'));
+        $this->assertTrue(in_array(0, $this->crudPanel->getOperationSetting('pageLengthMenu')[0]));
+    }
+
+    /**
+     * Tests define paginator length with multi array [[20, 30, 40],['v', 't', 'q']].
+     *
+     * @group failing
```

Comment:
@promatik dunno if you know about it. But I will let you know here why this `@group failing`. 

Using this `@group` technique we can run specific tests instead of the whole testsuit. 

After defining the tests you want to run in the same group just `vendor/bin/phpunit --testdox --group=failing`

Reply:
Great @pxpm didn't know about it, I was using filter to run tests related to this PR.
`vendor/bin/phpunit --testdox --filter CrudPanelChangePaginatorLength`. It acts like a regex match, so it will match all the tests with that string.